### PR TITLE
Owlvit test fixes

### PR DESCRIPTION
**Pull request by alaradirik**
# What does this PR do?

- Fixes assertion error in slow OWL-ViT integration tests
- Fixes incompatible cpu/gpu test errors caused by `OwlViTForObjectDetection.normalize_grid_corner_coordinates()`
- Sets `attention_mask` input argument  as optional

### CODE REVIEW

File: tests/models/owlvit/test_modeling_owlvit.py [hunk 1]
```diff
@@ -91,6 +91,7 @@ def __init__(
 
     def prepare_config_and_inputs(self):
         pixel_values = floats_tensor([self.batch_size, self.num_channels, self.image_size, self.image_size])
+        pixel_values = pixel_values.to(torch_device)
```
Comment (NielsRogge) [inline tests/models/owlvit/test_modeling_owlvit.py:94]: No need for this line, `floats_tensor` will automatically already put it on `torch_device` :)

File: tests/models/owlvit/test_modeling_owlvit.py [hunk 2]
```diff
@@ -245,10 +245,12 @@ def __init__(
 
     def prepare_config_and_inputs(self):
         input_ids = ids_tensor([self.batch_size * self.num_queries, self.seq_length], self.vocab_size)
+        input_ids = input_ids.to(torch_device)
```
Comment (NielsRogge) [inline tests/models/owlvit/test_modeling_owlvit.py:248]: Same here, no need for this line

File: tests/models/owlvit/test_modeling_owlvit.py [hunk 3]
```diff
@@ -245,10 +245,12 @@ def __init__(
...
         input_mask = None
 
         if self.use_input_mask:
             input_mask = random_attention_mask([self.batch_size * self.num_queries, self.seq_length])
+            input_mask = input_mask.to(torch_device)
```
Comment (NielsRogge) [inline tests/models/owlvit/test_modeling_owlvit.py:253]: Same here

**Review by NielsRogge — APPROVED** [review]
Thanks for fixing!

File: tests/models/owlvit/test_modeling_owlvit.py [hunk 4]
```diff
@@ -770,21 +759,20 @@ def test_inference(self):
             torch.Size(
                 (
                     inputs.pixel_values.shape[0],
-                    inputs.input_ids.shape[0] * inputs.input_ids.shape[1] * inputs.pixel_values.shape[0],
+                    inputs.input_ids.shape[0],
```
Comment (sgugger) [inline tests/models/owlvit/test_modeling_owlvit.py:762]: After the change the whole `torch.Size` fits in one line.

File: tests/models/owlvit/test_modeling_owlvit.py [hunk 5]
```diff
@@ -770,21 +759,20 @@ def test_inference(self):
...
                 )
             ),
         )
         self.assertEqual(
             outputs.logits_per_text.shape,
             torch.Size(
                 (
-                    inputs.input_ids.shape[0] * inputs.input_ids.shape[1] * inputs.pixel_values.shape[0],
+                    inputs.input_ids.shape[0],
```
Comment (sgugger) [inline tests/models/owlvit/test_modeling_owlvit.py:770]: Same here.

**Review by sgugger — APPROVED** [review]
Thanks for all the fixes!
